05/ukol7.py

```diff
@@ -0,0 +1,57 @@
+#Napiš tyto programy. Každý z nich bude mít na vstupu řetězec s rodným číslem a nějak ho
+#zanalyzuje:
+#(a) Je ve správném formátu: 6 číslice, lomítko, 4 číslice? (vypíše True nebo False)
+#(b) Je dělitelné jedenácti? (vypíše True nebo False)
+#(c) Jaké datum narození je v čísle zakódováno? (vypíše trojici čísel – den, měsíc, rok)
+#(d) Jaké pohlaví je v čísle zakódováno? (vypíše 'muž' nebo 'žena')
+#Pro účely úkolu stačí, když bude program umět zpracovat čísla vydávaná od roku 1985.
+#Reálná rodná čísla můžou být složitější :)
+
+
+
+#Osetreni vstupu
+while True:
+    pin = input('Zadej rodne cislo ve formatu xxxxxx/xxxx: ') 
+    if pin.replace('/','',1).isdigit():
+        break
+    print('Toto neni cislo ve formatu xxxxxx/xxxx!')
+
+#Pin musi mit 11 znaku, na seste pozici musi byt /, kdyz se lomitko na prvni pozici nahradi "", tak musi byt pin cislo.
+if len(pin) == 11 and pin[6] == '/' and pin.replace('/','',1).isdigit():
+    print('Kontrola formatu cisla: ', True)
+else:
+    print('Kontrola formatu cisla: ', False) 
+    exit(1)   
+
+pin = pin.replace('/','',1)         #Prvni pozici se znakem / vymenim za ""
+pin_int = int(pin)                  #Rodne cislo prevedu ze string na cislo
+result = pin_int / 11               #Rodne cislo vydelim 11 se zbytem
+
+if result == int(result):           #Vysledek musi byt cele cislo
```

Comment:
Tohle je asi taky způsob, jak to dělat, nicméně Python (a skoro snad každý jiný jazyk) má na tohle speciální [operátor `modulo`](https://docs.python.org/3.10/reference/expressions.html) `%`:

> The % (modulo) operator yields the remainder from the division of the first argument by the second.

Takže pokud výsledek modula _(např. `a % b`)_ rovná se nula, tak `a` je dělitelné `b`čkem beze zbytku.

Takže tenhle IF by mohl spíš raději být:

```python
if pin_int % 11 == 0:
  ...
```